### PR TITLE
Switch to launch template

### PR DESCRIPTION
**Pull request by peterdeme**
This is a contribution of @spr-mweber3 [here](https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/pull/28).

I have cherry picked his commits so that we can run CI/CD on the branch.

Additionally:
- Added a document generation mechanism

## Description

Launch Configuration deprecation scheduleThe annoucement from AWS provides the following key dates:

* **December 31, 2022** - No new EC2 features or new EC2 instance types will be added to launch configurations after this date.
* **March 31, 2023** - New accounts created after this date will not be able to create new launch configurations via the console. API and CLI access will remain available to support customers with automation use cases.
* **December 31, 2023** - New accounts created after this date will not be able to create new launch configurations.

https://aws.amazon.com/blogs/compute/amazon-ec2-auto-scaling-will-no-longer-add-support-for-new-ec2-features-to-launch-configurations/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
       source  = "hashicorp/aws"
-      version = "~> 4.6.0"
+      version = "~> 4.48.0"
     }
   }
 }
 
 module "my_workerpool" {
-  source = "github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2?ref=e954914020a5e1a561038cba27bb9d06438deba6"
+  source = "github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2"
```
Comment (peterdeme) [inline README.md:20]: We should start to use tags imo. In the next PR maybe?

File: README.md [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
 
   configuration = <<-EOT
     export SPACELIFT_TOKEN="${var.worker_pool_config}"
     export SPACELIFT_POOL_PRIVATE_KEY="${var.worker_pool_private_key}"
   EOT
 
-  max_size          = 1
-  ami_id            = var.worker_pool_ami_id
```
Comment (peterdeme) [inline README.md:26]: Ideally I want the customers to not specify this.

**Review by adamconnelly — APPROVED** [review]
Looks good @peterdeme and thanks for picking this up!

I'm not sure what your thoughts were for testing this? The idea I had was to temporarily update preprod to point at this branch, then we can merge this PR once we're happy the changes work correctly and don't give us any issues upgrading. What do you think?

**Comment by peterdeme** [comment]
@adamconnelly let's do that!

**Comment by peterdeme** [comment]
@adamconnelly I did a brief testing on my account btw: https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/pull/28#issuecomment-1360392246

**Comment by adamconnelly** [comment]
> @adamconnelly I did a brief testing on my account btw: [#28 (comment)](https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/pull/28#issuecomment-1360392246)

Awesome - looks good then.

**Pull request merged**